utils/gyb_syntax_support/Node.py

```diff
@@ -22,7 +23,24 @@ def __init__(self, name, description=None, kind=None, traits=None,
         self.description = description
 
         self.traits = traits or []
-        self.children = children or []
+        self.children = []
+        # Add implicitly generated GarbageNodes children in between any two 
+        # defined children
+        if kind != 'SyntaxCollection':
+            for i in range(2 * len(children) + 1):
```

Comment:
I agree that the trailing garbage is of very little value because most likely the current node doesn’t know until which node skip garbage tokens. I reduced the number of nodes to 2n.

Note that the placement of garbage nodes is still ambiguous because placement of nodes in `[garbage root 2]` and `[garbage bar 1]` results in the same syntax tree. I doubt this will be a problem in practice though. Whichever node knows that it wants to skip over tokens should place these in its garbage.